expression/util.go

```diff
@@ -259,6 +255,25 @@ func ColumnSubstituteImpl(expr Expression, schema *Schema, newExprs []Expression
 	return false, expr
 }
 
+func canSubstituted(coll, newFuncColl string) bool {
+	collGroupID, ok1 := CollationStrictnessGroup[coll]
+	newFuncCollGroupID, ok2 := CollationStrictnessGroup[newFuncColl]
+
+	if ok1 && ok2 {
+		if collGroupID == newFuncCollGroupID {
+			return true
+		}
+
+		for _, id := range CollationStrictness[collGroupID] {
+			if newFuncCollGroupID == id {
+				return true
+			}
+		}
+	}
+
+	return false
```

Comment:
I didn't see any difference with the original logic. The result will be exactly the same. 
Any examples or test cases to verify it?

Reply:
right, I did not change the logic. For now, every supported collation has clearly strictness-ship, original code works well. This PR will be useful in future if new collation added, such as `utf8mb4_unicode_ci`, There is no suitable strictness for `utf8mb4_unicode_ci`, because `utf8mb4_unicode_ci` and `utf8mb4_general_ci` cannot substituted each other. **In short, there is no parallel relationship in original code**

Reply:
Understood.